skimage/io/collection.py

```diff
@@ -72,38 +78,52 @@ def alphanumeric_key(s):
     return k
 
 
-class ImageCollection(object):
+def _is_multipattern(input_pattern):
+    """Helping function. Returns True if pattern contains a tuple, list, or a
+    string separated with os.pathsep."""
+    # Conditions to be accepted by ImageCollection:
+    has_str_ospathsep = (isinstance(input_pattern, str)
+                         and os.pathsep in input_pattern)
+    not_a_string = not isinstance(input_pattern, str)
+    has_iterable = isinstance(input_pattern, Sequence)
+    has_strings = all(isinstance(pat, str) for pat in input_pattern)
+
+    is_multipattern = (has_str_ospathsep or
+                       (not_a_string
+                        and has_iterable
+                        and has_strings))
+    return is_multipattern
 
-    """Load and manage a collection of image files.
 
-    Note that files are always stored in alphabetical order. Also note that
-    slicing returns a new ImageCollection, *not* a view into the data.
+class ImageCollection(object):
+    """Load and manage a collection of image files.
 
     Parameters
     ----------
-    load_pattern : str or list
-        Pattern glob or filenames to load. The path can be absolute or
-        relative.  Multiple patterns should be separated by os.pathsep,
-        e.g. '/tmp/work/*.png:/tmp/other/*.jpg'.  Also see
-        implementation notes below.
+    load_pattern : str or list of str
+        Pattern string or list of strings to load. The filename path can be
+        absolute or relative.
```

Comment:
@alexdesiqueira based on my reading of the notes from the last meeting, this description should include more detail, then we can merge!

Reply:
I'll point out that my approval still stands, and @stefanv could merge when ready. =)